lib/mungers/env.js

```diff
@@ -11,32 +11,36 @@ const colon = isWindows() ? ';' : ':'
  * Updates the environment variables to intercept `node` commands and pass down options.
  *
  * @param workingDir {string} Absolute system-dependent path to the directory containing the shim files.
- * @param options {import("../munge").InternalSpawnOptions} Internal spawn options to update.
- * @return {void} This function does not return any value, the options are modified in-place.
+ * @param options {import("../munge").InternalSpawnOptions} Original internal spawn options.
+ * @return {import("../munge").InternalSpawnOptions} Updated internal spawn options.
  */
 function mungeEnv(workingDir, options) {
+  const newEnvPairs = [...options.envPairs];
+
   let pathEnv
   for (let i = 0; i < options.envPairs.length; i++) {
```

Comment:
I'd prefer the copy variable be named `envPairs` so we can use short object notation for the final return `{...options, envPairs}`.

Side note I feel like this for loop could be rewritten to use `Array#map`:
```js
const envPairs = options.envPairs.map(ep => {
  // return value calculated from ep
})
```